aptos-move/aptos-gas/src/gas_meter.rs

```diff
@@ -203,10 +207,19 @@ impl InitialGasSchedule for AptosGasParameters {
     }
 }
 
+pub trait AptosGasMeter: MoveGasMeter {
+    fn balance(&self) -> Gas;
+    fn balance_internal(&self) -> InternalGas; // TODO: put this in a sealed trait.
+
+    fn charge_intrinsic_gas_for_transaction(&mut self, txn_size: NumBytes) -> VMResult<()>;
+
+    fn charge_write_item_gas(&mut self, key: &StateKey, op: &WriteOp) -> VMResult<()>;
```

Comment:
nit: should the public functions of `StandardGasMeter` also be public functions on the `AptosGasMeter` trait?

for example, having `change_set_configs` be available to call in `failed_transaction_cleanup` will save having to add a new argument to the method.

Reply:
Generally, I agree with what you said and this is what I ended up doing in the latest update. However `change_set_configs` is a bit different -- it's cleaner logic-wise to have the Aptos VM, rather than the gas meter to carry it.